skll/data/readers.py

```diff
@@ -327,6 +340,22 @@ def _parse_dataframe(self, df, id_col, label_col, features=None):
             raise ValueError("No features found in possibly "
                              "empty file '{}'.".format(self.path_or_list))
 
+        if drop_blanks and replace_blanks_with is not None:
+            raise ValueError("You cannot both drop blanks and replace them. "
+                             "'replace_blanks_with' can only have a value when "
+                             "'drop_blanks' is `False`.")
+
+        # should we replace blank values with something?
+        if replace_blanks_with is not None:
+            self.logger.info('Blank values in all rows/lines will be replaced with '
+                             'user-specified value(s).')
+            df = df.fillna(replace_blanks_with)
+
+        # should we remove lines that have any NaNs?
+        if drop_blanks:
+            self.logger.info('Rows/lines with any blank values will be dropped.')
+            df = df.dropna().reset_index(drop=True)
```

Comment:
`df.dropna(inplace=True)` does equivalent to this, and looks cleaner.

Reply:
Ok, it does not reset index. but is it necessary.

Reply:
Yes, I think it is because otherwise the row numbers are not sequential.